fastschema.go

```diff
@@ -193,8 +220,30 @@ func (a *App) Reload(ctx context.Context, migration *db.Migration) (err error) {
 
 // UpdateCache updates the application cache.
 // It fetches all roles from the database and stores them in the cache.
-func (a *App) UpdateCache(ctx context.Context) (err error) {
-	if a.roles, err = db.Query[*fs.Role](a.DB()).Select(
+func (a *App) UpdateCache(ctx context.Context, keys ...string) (err error) {
+	if len(keys) > 0 {
+		for _, key := range keys {
+			if key == "roles" {
+				err = a.UpdateRoleCache(ctx)
+
+				if err != nil {
+					panic(err)
+					return err
+				}
+			}
+		}
+
+		return nil
+	}
+
+	// if keys is empty, update all cache such as role...
+	err = a.UpdateRoleCache(ctx)
+
+	return err
```

Comment:
You can try referring to this way of writing:

```suggestion
if len(keys) == 0 {
   // if keys is empty, update all cache such as role...
   return a.UpdateRoleCache(ctx)
}

for _, key := range keys {
    if key == "roles" {
        if err := a.UpdateRoleCache(ctx); err != nil {
            panic(err)
         }
        break
    }
}

return nil
```